### PR TITLE
feat: event signup and optout deadline

### PR DESCRIPTION
**Pull request by ipeglin**
## Changes
* Events with a capacity now has to declare a signup time window. Outside this window, a user cannot sign up or opt out of events.
* Admins can bypass this by manually registering specific users at the edit event page.

### CODE REVIEW

**Review by JStoresund — APPROVED** [review]
LGTM

**Review by AasmundN — CHANGES_REQUESTED** [review]
An event page must display when the registration opens and closes.

File: pages/event/[id].vue [hunk 1]
```diff
@@ -61,6 +61,33 @@
       event.value.attendants.length >= event.value.capacity
   )
 
+  // does the event have any registration actions, and is user signed in
+  const hasEventActions = computed(() => useAuth.isLoggedIn && hasCapacity)
+
+  // check if registration actions should be rendered on client side
+  const showRegistrationAction = ref(false)
```
Comment (AasmundN) [inline pages/event/[id].vue:68]: This needs to be modified in order to show the button correctly on SSR.

**Review by AasmundN — COMMENTED** [review]
When you are registered for the event and the registration closes you still need to be able to see whether or not you are registered, now it just says that registration is not open.

**Comment by AasmundN** [comment]
Admins shuld be able to register users (both their own user and other users) to an event. We should add a input field for entering a user id for registration on the event edit page.

**Comment by AasmundN** [comment]
I added so that admins can register users to events at will.

**Review by AasmundN — APPROVED** [review]

**Review by JStoresund — APPROVED** [review]
LGTM!

**Pull request merged**